agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/InternalJarURLHandler.java

```diff
@@ -75,12 +102,32 @@ protected URLConnection openConnection(final URL url) throws IOException {
       // nullInputStream() is not available until Java 11
       return new InternalJarURLConnection(url, new ByteArrayInputStream(new byte[0]));
     }
-    final JarEntry entry = filenameToEntry.get(filename);
-    if (null != entry) {
-      return new InternalJarURLConnection(url, bootstrapJarFile.getInputStream(entry));
+    // believe it or not, we're going to get called twice for this,
+    // and the key will be a new object each time.
+    Pair<String, JarEntry> pair = cache.get();
+    if (null == pair || !filename.equals(pair.getLeft())) {
+      StringBuilder sb = JAR_ENTRY_QUERY.get();
```

Comment:
Just curious, is it possible for this class to be accessed from multiple threads? I wonder how many, if there are a lot of copies of this buffer, it's unfortunate they all get stuck in memory after classloading is done sort of lean towards not using a threadlocal for it, string seems pretty small.

Reply:
I sort of lean towards not using the threadlocal optimization for these small strings and let it all be garbage that can be collected, over the complexity of weak references, since I suspect this is a tiny part of the optimization in the attached PR.

But if we're only leaking a few hundred bytes it's not so bad either.

Reply:
@trask actually this was an insurance policy against there ever being more than _one_ thread accessing the variable. We haven't seen any negative consequences in our benchmarks, but I think it's a question of "take it or leave it" - if it doesn't look right to you, please just change it.

Reply:
removed the `ThreadLocal<StringBuilder>`